File: LC/lc461/lc461.py
class Solution:
    def hammingDistance(self, x: int, y: int) -> int:
        if x == y:
            return 0
        if x% 2 == y %2: 
            return self.hammingDistance(x>>1, y>>1)
        return 1+ self.hammingDistance(x>>1, y>>1)

File: LC/lc461/lc461.js
/**
 * @param {number} x
 * @param {number} y
 * @return {number}
 */
var hammingDistance = function(x, y) {
  var res = 0;
  for (i = 0; i <31; i++){
      if (x%2 - y%2 !==0){
          res += 1;
      }
      x >>= 1;
      y >>= 1;
  }
  return res;
};